mitosheet/mitosheet/enterprise/mito_config.py

```diff
@@ -75,21 +120,44 @@ def __init__(self):
         if self.mec is not None:
             log('loaded_mito_enterprise_config')
 
-    def _get_version(self) -> str:
+    def get_version(self) -> str:
         if self.mec is None or self.mec[MITO_CONFIG_KEY_VERSION] is None:
-            return '1' # NOTE: update this to be the most recent version, when we bump the version
+            return '2' # NOTE: update this to be the most recent version, when we bump the version
         return self.mec[MITO_CONFIG_KEY_VERSION]
 
-    def _get_support_email(self) -> str:
+    def get_support_email(self) -> str:
         if self.mec is None or self.mec[MITO_CONFIG_KEY_SUPPORT_EMAIL] is None:
             return DEFAULT_MITO_CONFIG_SUPPORT_EMAIL
         return self.mec[MITO_CONFIG_KEY_SUPPORT_EMAIL]
 
+    def get_code_snippets_version(self) -> Optional[str]:
+        if self.mec is None or self.mec[MITO_CONFIG_KEY_CODE_SNIPPETS_VERSION] is None:
+            return None
+        return self.mec[MITO_CONFIG_KEY_CODE_SNIPPETS_VERSION]
+
+    def get_code_snippets_url(self) -> Optional[str]:
+        if self.mec is None or self.mec[MITO_CONFIG_KEY_CODE_SNIPPETS_URL] is None:
+            return None
+        return self.mec[MITO_CONFIG_KEY_CODE_SNIPPETS_URL]
+
+    def get_code_snippets_support_email(self) -> str:
+        if self.mec is None or self.mec[MITO_CONFIG_KEY_CODE_SNIPPETS_SUPPORT_EMAIL] is None:
+            return DEFAULT_MITO_CONFIG_CODE_SNIPPETS_SUPPORT_EMAIL
+        return self.mec[MITO_CONFIG_KEY_CODE_SNIPPETS_SUPPORT_EMAIL]
+
+    def get_code_snippets(self) -> Dict[str, Optional[str]]:
+        return {
+            MITO_CONFIG_KEY_CODE_SNIPPETS_VERSION: self.get_code_snippets_version(),
+            MITO_CONFIG_KEY_CODE_SNIPPETS_URL: self.get_code_snippets_url(),
+            MITO_CONFIG_KEY_CODE_SNIPPETS_SUPPORT_EMAIL: self.get_code_snippets_support_email(),
+        }
```

Comment:
Let me try and be really specific here -- I think we've got back and forth a few times on this request. 

The type of this function should be Optional[Dict[str, str]]. That is, if all of the 3 variable are defined, then we get all of the objects back. If any of them are not defined, then we get None back. 

This allows us to enforce, in the type system, that the user is providing _all three_ required variables for the code snippets - and makes consumption of the config easier throughout the codebase!

Reply:
I think this would be further improved by:
1. Removing the accessors for getting the version or URL individually (or at least underscoring them to make them private). Again, relying more on the type system here. 
2. Using a TypedDict (see types.py) -- and actually being clear what is in the dictonary. 

These two things together will result in a _very nice_ config API. I think having a nice config API is pretty important, as we will have a combinatorial explosion of configuration options eventually. Protecting ourselves from ourselves is going to be really important here!